chore: Refine Custom Network button and avatar styles

amandaye0h opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

<!--
Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:
1. What is the reason for the change?
2. What is the improvement/solution?
-->

## **Changelog**

<!--
If this PR is not End-User-Facing and should not show up in the CHANGELOG, you can choose to either:
1. Write `CHANGELOG entry: null`
2. Label with `no-changelog`

If this PR is End-User-Facing, please write a short User-Facing description in the past tense like:
`CHANGELOG entry: Added a new tab for users to see their NFTs`
`CHANGELOG entry: Fixed a bug that was causing some NFTs to flicker`

(This helps the Release Engineer do their job more quickly and accurately)
-->

CHANGELOG entry: chore: Refine Custom Network button and avatar styles

## **Related issues**

Fixes: https://consensyssoftware.atlassian.net/jira/software/c/projects/MDP/boards/2972?search=custom&selectedIssue=MDP-242

## **Manual testing steps**

```gherkin
Feature: my feature name

  Scenario: user [verb for user action]
    Given [describe expected initial app state]

    When user [verb for user action]
    Then [describe expected outcome]
```

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

### **After**

<img width="793" height="786" alt="Screenshot 2025-12-03 at 1 02 59 PM" src="https://github.com/user-attachments/assets/f4f95bf3-eb9d-427b-9d6f-d448a383f415" />


## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> Centers cell row content, updates Custom Network selector styling (button design, themed styles), and increases network avatar size; adjusts tests and snapshots accordingly.
> 
> - **UI**:
>   - **Cells**: Center-align row content by adding `alignItems: 'center'` in `components-temp/CellSelectWithMenu/CellSelectWithMenu.styles.ts`.
>   - **Custom Network Selector**:
>     - Refactor `CustomNetworkSelector.styles.ts` to accept themed params and update styles (rounded/padded `addNetworkButtonContainer`, new `iconContainer` with muted background, 32x32, centered).
>     - Update footer button: wrap `Add` icon in `iconContainer`, switch icon/text to primary color, text variant to `BodyMDMedium`.
>     - Increase avatar size in list items to `AvatarSize.Md`.
> - **Tests/Snapshots**:
>   - Update `useStyles` call expectation to use `createStyles` with `{}`.
>   - Refresh snapshots to reflect centered rows and updated layouts.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 605ea1404eafcc402391f727c6483c2e11e7476f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

amandaye0h committed:
chore: refine button and avatar styles

Prithpal-Sooriya committed:
Merge branch 'main' into custom-network/dqa

amandaye0h committed:
fix: update CustomNetworkSelector test to match current component behavior

The component now passes an empty object to useStyles instead of colors.
Updated the test assertion to match the current implementation.

Prithpal-Sooriya commented:
@cursor fix failing unit tests and please use the PR author for the commit (and future commits)

amandaye0h committed:
fix: update CellSelectWithMenu snapshots for alignItems style change

Updated snapshots to reflect the addition of alignItems: 'center' to
the cellBase style in CellSelectWithMenu component.
